pkg/relayer/interface.go

```diff
@@ -162,3 +163,18 @@ type SessionTree interface {
 	// GetTrieSpec returns the trie spec of the SMST.
 	GetTrieSpec() smt.TrieSpec
 }
+
+// RelayMeter is an interface that keeps track of the consumed stake for each application.
+type RelayMeter interface {
+	// Start starts the relay meter.
+	Start(ctx context.Context) error
+	// AccumulateRelayReward accumulates the relay reward for the given relay request.
+	// The relay cost is added optimistically, assuming that the relay will be volume / reward
+	// applicable and the relay meter would remain up to date.
+	AccumulateRelayReward(ctx context.Context, relayRequestMeta servicetypes.RelayRequestMetadata) error
```

Comment:
Can you #PUC why we're going with this `OptimisticAccumulate` + `SetNonApplicable` approach rather than just `AccumulateVolumeApplicableRelay`.

The latter seems simpler but I"m not quite following.